shared-libs/lineage/src/hydration.js

```diff
@@ -358,6 +359,31 @@ module.exports = function(Promise, DB) {
       });
   };
 
+  const fetchHydratedDocs = docIds => {
+    if (!Array.isArray(docIds) || !docIds.length) {
+      return Promise.resolve([]);
+    }
```

Comment:
I'm not sure about resolving [] when given rubbish input. In this case I think it's clear that the developer made a mistake and didn't follow the API, so I think it's more correct to reject, to make it easy to find and debug these mistakes.

```suggestion
    if (!Array.isArray(docIds) {
      return Promise.reject(new Error('Invalid parameter: "docIds" must be an array'));
    }
    if (!docIds.length) {
      return Promise.resolve([]);
    }
```

What do you think?